cub/cub/device/device_segmented_reduce.cuh

```diff
@@ -34,6 +34,10 @@
 
 #include <cub/config.cuh>
 
+#include "cuda/std/detail/libcxx/include/__iterator/readable_traits.h"
+#include "cuda/std/detail/libcxx/include/__type_traits/integral_constant.h"
+#include "cuda/std/detail/libcxx/include/__type_traits/is_integral.h"
```

Comment:
I believe for now we went with the idea that cub should mostly include the public headers from libcu++

We might change that but until then I would recommend to include just `<cuda/std/type_traits>` and `<cuda/std/iterator>`

Also those headers are probably added by clangd because they are in front of our system header guard. We want those to be after it